javapoet/src/main/java/com/palantir/javapoet/ParameterizedTypeName.java

```
@@ -43,7 +43,8 @@ private ParameterizedTypeName(
             List<TypeName> typeArguments,
             List<AnnotationSpec> annotations) {
         super(annotations);
-        this.rawType = checkNotNull(rawType, "rawType == null").annotated(annotations);
+        this.rawType =
+                checkNotNull(rawType, "rawType == null").withoutAnnotations().annotated(annotations);
```

It seems a bit wrong to create intermediary rawType multiple times.

I'm wondering whether we could
- Get rid of the constructor taking an annotations arg
- Have the rawType be provided with the annotations already (e.g. replace https://github.com/palantir/javapoet/blob/1c5573db4d721ceac9192f567373030df0225846/javapoet/src/main/java/com/palantir/javapoet/ParameterizedTypeName.java#L71-L73 by `new ParameterizedTypeName(enclosingType, rawType.annotated(annotations), typeArguments);`
- Get the annotations from `rawType.annotations()` to pass to `super`

That constructor seems to only ever be called (with a non-empty list anyway) from that one place

---

Thanks for the suggestion, this is much nicer.

This actually revealed and fixed a separate bug where

```java
ParameterizedTypeName get(ClassName rawType, TypeName... typeArguments)
```

would silently drop any annotations that on `rawType`.

I've added a test to verify this, now fixed, behavior.